src/module_3/module3_statistical_learning.md

```diff
@@ -0,0 +1,1368 @@
+            <Axes: title={'center': 'abandoned_before'}>,
+            <Axes: title={'center': 'active_snoozed'}>],
+           [<Axes: title={'center': 'set_as_regular'}>,
+            <Axes: title={'center': 'normalised_price'}>,
+            <Axes: title={'center': 'discount_pct'}>,
+            <Axes: title={'center': 'global_popularity'}>,
+            <Axes: title={'center': 'count_adults'}>],
+           [<Axes: title={'center': 'count_children'}>,
+            <Axes: title={'center': 'count_babies'}>,
+            <Axes: title={'center': 'count_pets'}>,
+            <Axes: title={'center': 'people_ex_baby'}>,
+            <Axes: title={'center': 'days_since_purchase_variant_id'}>],
+           [<Axes: title={'center': 'avg_days_to_buy_variant_id'}>,
+            <Axes: title={'center': 'std_days_to_buy_variant_id'}>,
+            <Axes: title={'center': 'days_since_purchase_product_type'}>,
+            <Axes: title={'center': 'avg_days_to_buy_product_type'}>,
+            <Axes: title={'center': 'std_days_to_buy_product_type'}>],
+           [<Axes: title={'center': 'order_size'}>, <Axes: >, <Axes: >,
+            <Axes: >, <Axes: >]], dtype=object)
+
+
+
+
+    
+![png](module3_statistical_learning_files/module3_statistical_learning_8_1.png)
+    
+
+
+
+```python
+# Correlation matrix
+
+columns = numerical_cols + label_col
+
+def plot_correlation_matrix(df: pd.DataFrame, columns: list):
+    # Compute the correlation matrix
+    corr = df[columns].corr()
+
+    # Generate a mask for the upper triangle
+    mask = np.triu(np.ones_like(corr, dtype=bool))
+
+    # Set up the matplotlib figure
+    fig, ax = plt.subplots (figsize=(11, 9))
+
+    # Generate a custom diverging colormap
+    cmap = sns.diverging_palette (230, 20, as_cmap=True)
+
+    # Draw the heatmap with the mask and correct aspect ratio
+    sns.heatmap(corr, mask=mask, cmap=cmap, vmax=.3, center=0,# annot=True,
+            square=True, linewidths=.5, cbar_kws={"shrink": .5})
+    
+plot_correlation_matrix(df_orders, columns)
+```
+
+
+    
+![png](module3_statistical_learning_files/module3_statistical_learning_9_0.png)
+    
+
+
+## Create train and test sets
+
+Puntos a tener en cuenta
+- Hacer el split por order_id - No puede haber un order_id en train y test
+- Debemos tener en cuenta las fechas -  Ventas futuras de un mismo usuario se verán influenciadas por ventas pasadas (producto previamente en abandoned_cart o si un producto se ha comprado recientemente)
+
+
+```python
+# Before splitting the data we'll handle datetimes
+
+### Extract components of datetimes
+df_orders = df_orders.drop('created_at', axis=1) # I'll only keep order_date as they display almost the same info
+
+df_orders['order_year'] = df_orders['order_date'].dt.year
+df_orders['order_month'] = df_orders['order_date'].dt.month
+df_orders['order_day_of_week'] = df_orders['order_date'].dt.dayofweek 
+
+df_orders = df_orders.drop('order_date', axis=1)
```

Comment:
what's the point of dropping columns?

Reply:
I think here I tried to replace the date by year, month and day ints because the model didn't let me input dates